controllers/storagecluster/cephrbdmirrors.go

```diff
@@ -0,0 +1,158 @@
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type ocsCephRbdMirrors struct{}
+
+func (r *StorageClusterReconciler) deleteCephRbdMirrorInstance(cephRbdMirror *cephv1.CephRBDMirror, existing *cephv1.CephRBDMirror) error {
+	var err error
+
+	if cephRbdMirror.GetDeletionTimestamp().IsZero() {
+		r.Log.Info("Deleting CephRbdMirror.", "CephRbdMirror", klog.KRef(cephRbdMirror.Namespace, cephRbdMirror.Name))
+		err = r.Client.Delete(context.TODO(), existing)
+	}
+	return err
+}
+
+// newCephRbdMirrorInstances returns the cephRbdMirror instances that should be created
+// on first run.
+func (r *StorageClusterReconciler) newCephRbdMirrorInstances(initData *ocsv1.StorageCluster) ([]*cephv1.CephRBDMirror, error) {
+
+	var rbdMirroringSpec cephv1.RBDMirroringSpec
+	rbdMirroringSpec.Count = 1
+	rbdMirroringSpec.Resources = corev1.ResourceRequirements{
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1"),
+			corev1.ResourceMemory: resource.MustParse("2Gi"),
+		},
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("1"),
+			corev1.ResourceMemory: resource.MustParse("2Gi"),
+		},
+	}
+
+	ret := []*cephv1.CephRBDMirror{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      generateNameForCephRbdMirror(initData),
+				Namespace: initData.Namespace,
+			},
+			Spec: rbdMirroringSpec,
+		},
+	}
+	for _, obj := range ret {
+		err := controllerutil.SetControllerReference(initData, obj, r.Scheme)
+		if err != nil {
+			r.Log.Error(err, "Unable to set controller reference for CephRBDMirror.", "CephRBDMirror", klog.KRef(obj.Namespace, obj.Name))
+			return nil, err
+		}
+	}
+	return ret, nil
+}
+
+// ensureCreated ensures that cephRbdMirror resources exist in the desired
+// state.
+func (obj *ocsCephRbdMirrors) ensureCreated(r *StorageClusterReconciler, instance *ocsv1.StorageCluster) error {
+	reconcileStrategy := ReconcileStrategy(instance.Spec.ManagedResources.CephRbdMirrors.ReconcileStrategy)
+	if reconcileStrategy == ReconcileStrategyIgnore {
+		return nil
+	}
+
+	cephRbdMirrors, err := r.newCephRbdMirrorInstances(instance)
+	if err != nil {
+		return err
+	}
+
+	for _, cephRbdMirror := range cephRbdMirrors {
+		existing := cephv1.CephRBDMirror{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{Name: cephRbdMirror.Name, Namespace: cephRbdMirror.Namespace}, &existing)
+		switch {
+		case !instance.Spec.Mirroring.Enabled:
+			if err == nil {
+				err = r.deleteCephRbdMirrorInstance(cephRbdMirror, &existing)
+				if err != nil {
+					r.Log.Error(err, "Failed to delete CephRbdMirror.", "CephRbdMirror", klog.KRef(existing.Namespace, existing.Name))
+					return fmt.Errorf("Failed to delete CephRbdMirror %v: %v", existing.Name, err)
+				}
```

Comment:
If CephRBDMirror is not present,this deletion will fail and error out. We need to handle such errors.